src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -116,79 +118,92 @@ func logHostsStatus(log logrus.FieldLogger, hosts map[string]inventory_client.Ho
 	log.Infof("Hosts status: %v", hostsStatus)
 }
 
-func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
+func (c *controller) WaitAndUpdateNodesStatus(ctx context.Context, wg *sync.WaitGroup) {
+	approveCtx, approveCancel := context.WithCancel(ctx)
```

Comment:
handling CSR within another go routine is very confusing and hard to maintain. This should be a goroutine fired from main() and the way to handle multiple context cancel is with the cancel listener

Reply:
it will make much more complex to maintain it from main. This routine is relevant to WaitAndUpdateNodesStatus and depends on it. When WaitAndUpdateNodesStatus exits, i want to exit this one. I think it is better to have it here. Not sure why it is bad to call go routine from function?